File: aula02-unmanaged-static-link/PontoLib/Ponto.cpp
// Ponto.cpp : Defines the exported functions by Ponto.lib
//

#include "Ponto.h"
#include <math.h>
#include <iostream>

/*
 * This is the constructor of a class that has been exported.
 * see aula01_Ponto.h for the class definition
 */
Ponto::Ponto(int x, int y)
{
	this->_z = 0;
	this->_x = x;
	this->_y = y;
}

double Ponto::getModule() {	 
	return sqrt((double)_x*_x + _y*_y + _z*_z);
}

void Ponto::print(){
	printf("Point v2 (x = %d, y = %d, z=%d)\n", _x, _y, _z);
}

File: aula13-const-readonly/Ponto.cs
using System;

public class Ponto{

	public int _x, _y;

	public Ponto(int x, int y)
	{
		this._x = x;
		this._y = y;
	}

    public int X{
        get{return _x;}
        set{
            if(value < 0)
                throw new ArgumentException ();
            _x = value;
        }
    }
    
	public double Module {	 
		get{
            return Math.Sqrt((double)_x*_x + _y*_y);
        }
	}

	public void Print(){
		Console.WriteLine("Version 2 Point (x = {0}, y = {1}))", _x, _y);
	}

    static void Main(){}
}